### ISSUE TITLE
Allow a query to sort based on an object value field

### ISSUE THREAD

**Issue by acinader** [parse-community/parse-server#4805]
Currently, if one tries to sort a query on a field in an object field, parse generates an error:

```
Type: Parse\ParseException
Code: 105
Message: Invalid field name: trendRankTopic.popular.
```

due to the check in the database controller here: https://github.com/parse-community/parse-server/blob/3b40d2227d9b9b7a768de0c6faf6916982fe8e89/src/Controllers/DatabaseController.js#L903

This is not the same, though related to: #2113 

An example would be an object:

` { sortField: { value: 10 } }`

with a Query:

`new Parse.Query('Test').addAscending('sortField.value').first()`

generates the above error

**Comment by flovilmart** [comment]
What error would do you expect? Or would you expect sorting to work on objects?

**Comment by acinader** [comment]
one sec and I'll open the pr with the failing test and my proposed solution :).

**Comment by flovilmart** [comment]
Even better :)

**Comment by acinader** [comment]
@flovilmart any thoughts on #4806 

I'm currently running our production on that branch which ain't a great long-term solution.

Anything I can explain as to why its an issue in the first place
or the solution I chose and why.

Some things I'd appreciate your considering: Is there a use case that this change would cause a breakage?  is there a use case where my change will allow an introduction of 'bad schema' that could cause mayhem?  Is there coverage I could write to test?

And finally, any suggestion on who I might ping to look at why Postgress is failing my test and what a solution might be?

**Comment by flovilmart** [comment]
I don’t think it’s a bad idea to introduce sorting on nested keys, Postgres support could come later. It can be gated with it_only_db instead of a simple it.

I don’t believe it is something we may regret in then future, however, I’m quite not understanding the change / i believe it’s not the proper resolution :)

**Comment by acinader** [comment]
I'll gate the test and add more explanation to the pull request.  I agree that it is low risk.  Having looked at it, I do think its the right solution, however, I'm super mindful of the fact that my ability to reason about that code is pretty limited :). Thanks!

**Comment by flovilmart** [comment]
CAN we add a small comment as for the change then in the code? So anyone that stumbled upon it is able to get a good idea of what’s going on?

**Comment by acinader** [comment]
done.

**Comment by wret** [comment]
Is it fixed on version 2.8.2?

**Comment by NicksonYap** [comment]
@wret 
It is enabled in 2.8.4: "Allow sorting an object field (#4806)"
https://github.com/parse-community/parse-server/blob/2.8.4/CHANGELOG.md

However, it does not seem like it's supposed to order by related object's field

Tried this and has totally no effect:
```
      query.include('subzone');
      query.include('subzone.zone');
      query.addAscending('subzone.zone.name')
      query.addAscending('subzone.name')
      query.addAscending('name')
```
